[hotfix][docs] Add Kafka-to-Kafka SQL example to documentation

Ashfaqbs opened the pull request:
## What is the purpose of the change

This is a documentation-only change, submitted as a `[hotfix][docs]` per contribution guidelines.

It adds an example demonstrating how to use Apache Flink SQL to build an end-to-end streaming pipeline using Kafka as both the source and the sink.

The example illustrates:
- Defining input and output Kafka tables using Flink SQL.
- Writing a transformation query that filters and modifies the stream.
- Setting up Kafka topics and verifying the results.
- Viewing the job graph in the Flink UI.

## Brief change log

- Added `kafka-to-kafka.md` under `docs/content/docs/dev/table/sql/examples/`
- Includes sample SQL statements, Kafka commands, and notes for verifying output.

## Does this pull request potentially affect one of the following parts:

- Dependencies: No
- The public API: No
- The documentation: **Yes**
- The build system: No
- The runtime per-job behavior: No

## Documentation

Adds a new SQL usage example under:
`docs/dev/table/sql/examples/kafka-to-kafka.md`

Ashfaqbs committed:
docs: add Kafka-to-Kafka SQL example using Flink SQL

This documentation adds a step-by-step example demonstrating how to
consume JSON messages from one Kafka topic, process them with Flink SQL,
and write the transformed results to another Kafka topic.

Includes producer/consumer steps, Flink SQL statements, and Flink UI visualization.
The example targets users looking to build end-to-end streaming pipelines using SQL.

davidradl commented on docs/content/docs/dev/table/sql/examples/kafka-to-kafka/kafka-to-kafka.md:
```diff
@@ -0,0 +1,106 @@
+  --partitions 1 --replication-factor 1
+
+# Create output topic
+docker exec -it docker-kafka-1 kafka-topics.sh \
```
- I see the Kafka script is run in Flink using:
 $KAFKA_DIR/bin/kafka-topics.sh 
do we not need to fully qualify it?
- can we tell the user where to get the Kafka docker image from.

Ashfaqbs replied:
Yes, makes sense. I've added the full path /opt/kafka/bin/kafka-topics.sh in the Docker commands.

davidradl commented on docs/content/docs/dev/table/sql/examples/kafka-to-kafka/kafka-to-kafka.md:
````diff
@@ -0,0 +1,106 @@
+);
+```
+
+## Submit SQL Job
````
I suggest pointing out explicitly that the UPPER is the transformation and the where clause its the filter.

Ashfaqbs replied:
Acknowledged. I’ve annotated the SQL query and also highlighted it in the summary section for clarity.

Ashfaqbs committed:
Merge branch 'apache:master' into patch-1

Ashfaqbs committed:
docs: refine Kafka to Kafka SQL example based on PR review feedback

- Replaced Windows-specific paths with platform-neutral or Unix-style commands
- Renamed `i-topic` and `o-topic` to `input-topic` and `output-topic` for clarity
- Added full path to `kafka-topics.sh` to avoid ambiguity in Docker usage
- Linked to Kafka Docker image sources for user reference
- Explicitly explained transformation (`UPPER`) and filter (`WHERE`) in SQL
- Improved formatting, clarity, and robustness throughout the markdown

Ashfaqbs commented on docs/content/docs/dev/table/sql/examples/kafka-to-kafka/kafka-to-kafka.md:
````diff
@@ -0,0 +1,106 @@
+Run the following Python script to send JSON messages to the `i-topic`.
+
+```bash
+C:\tmp\flink-sql\kafka-scripts>python producer.py
````
Thanks David, good point. I've replaced the Windows shell prompt with platform-neutral commands and removed the Windows-specific paths.

Ashfaqbs commented on docs/content/docs/dev/table/sql/examples/kafka-to-kafka/kafka-to-kafka.md:
````diff
@@ -0,0 +1,106 @@
+```bash
+# Create input topic
+docker exec -it docker-kafka-1 kafka-topics.sh \
+  --create --topic i-topic \
````
Agreed — updated the topic names to input-topic and output-topic for clarity.

Ashfaqbs commented:
@davidradl, Updated PR title to [hotfix][docs] … and clarified in the description that this is a documentation-only change per the contribution guidelines.
